File: leetcode/python/409_longest_palindrome.py
from typing import Counter


class Solution:
    def longestPalindrome(self, s: str) -> int:
        length: int = 0
        counter = Counter(s)
        hasCenter = False
        for c in counter.keys():
            
            if counter[c] & 1 == 0:
                print(c, counter[c])
                length += counter[c]
            else:
                length += counter[c] - 1
                hasCenter = True
        return length + (1 if hasCenter else 0)

s = Solution()
# print(s.longestPalindrome("abccccdd"))
# print(s.longestPalindrome("a"))
print(s.longestPalindrome("bb"))

File: leetcode/rust/src/409_longest_palindrome.rs
use std::collections::HashSet;

impl Solution {
    pub fn longest_palindrome(s: String) -> i32 {
        let mut count = 0;
        let mut set = HashSet::new();
        for c in s.chars() {
            if set.contains(&c) {
                set.remove(&c);
                count += 2;
            } else {
                set.insert(c);
            }
        }
        if !set.is_empty() {
            count += 1;
        }
        count
    }
}
